template/src/api/index.ts

```diff
@@ -0,0 +1,14 @@
+import {
+  GenericActionWithParameters as Action,
+  GenericActionWithoutParameters as ActionNoParams
+} from 'moleculer-service-ts';
+import { WelcomeParams } from './params/welcome.params';
```

Comment:
This means that when we hand out our interfaces to other teams developing other services, we have to include multiple files instead of a single file. So, we really need to package into a npm package that other services can import.

Reply:
@ujwal-setlur @ambuznego 
below you have `export * from './params/<file_names>` statements - we have to add them when exporting api.

To import api of `micro-discount` in other service it is sufficient to do:
`import * as DiscountApi from 'micro-discount'` as npm will search `package.json` for `main` property.

Currently our main is pointing to **src/api** - we do not have npm repo so we will install micro-service dependencies via url:
```
"dependencies": {
    "micro-discount": "git+ssh://git@github.com/visionmedia/express.git"
```
If our micro-services would be on npm we should point `main` to **dist/lib** (instead of **src**) as npm should contain only files that are built.

Reply:
OK, I like this approach.